### PR TITLE
spec: There is no docker by default in Fedora 24

### PR DESCRIPTION
**Pull request by stefwalter**
Unfortunately there is going to be no docker by default in Fedora 24
and later. This means that we shouldn't depend on 'docker' from deps
of our 'cockpit' meta package on Fedora 24.

### CODE REVIEW

File: tools/cockpit.spec [hunk 1]
```diff
@@ -87,11 +87,12 @@ Requires: %{name}-networkmanager = %{version}-%{release}
...
 Requires: %{name}-shell = %{version}-%{release}
 Requires: %{name}-storaged = %{version}-%{release}
+
+# Fedora does not want docker in its default install after F24
+%if 0%{?fedora} == 0 || 0%{?fedora} < 24
 %ifarch x86_64 armv7hl
 Requires: %{name}-docker = %{version}-%{release}
 %endif
-%if 0%{?rhel}
-Requires: %{name}-subscriptions = %{version}-%{release}
```
Comment (mvollmer) [inline tools/cockpit.spec]: Why has this been removed?

Comment (stefwalter) [inline tools/cockpit.spec]: Because it's not necessary. The rhel cockpit-shell includes subscriptions support and doesn't need to depend on it. Do you want me to make this a separate commit?

Comment (mvollmer) [inline tools/cockpit.spec]: Yes, it's better to make this change more visible, with its own commit message.

**Comment by stefwalter** [comment]
FYI @sgallagher @simo5

**Comment by sgallagher** [comment]
I don't think this is strictly necessary. I modified comps.xml[1] to just pull in the non-docker Cockpit sub-packages explicitly rather than the Cockpit meta-package.

I think it would actually be better to leave cockpit-docker in the cockpit meta-package so it gets pulled in if someone manually installs that meta-package.

[1] https://git.fedorahosted.org/cgit/comps.git/commit/

**Comment by stefwalter** [comment]
> I don't think this is strictly necessary. I modified comps.xml[1] to just pull in the non-docker Cockpit sub-packages explicitly rather than the Cockpit meta-package.

> I think it would actually be better to leave cockpit-docker in the cockpit meta-package so it gets pulled in if someone manually installs that meta-package.

Okay that's a good plan. In addition, this will help somewhat with not pulling deps into Fedora without vetting them.

**Comment by sgallagher** [comment]
As an addendum, I think that since we're going to be carefully choosing the default-installed cockpit pieces, we may actually want to *increase* the set of components that are pulled in by the `cockpit` meta-package (possibly including `cockpit-pcp` and `cockpit-kubernetes` as Recommends: dependencies).

Or, phrased differently: have the `cockpit` meta-package represent the full set of capabilities that upstream considers stable, irrespective of the default installation.
